src/test/unzip.spec.ts

```diff
@@ -0,0 +1,165 @@
+import { expect } from "chai";
+import * as fs from "fs";
+import * as path from "path";
+import { unzip } from "../unzip";
+import { DownloadDetails } from "../emulator/downloadableEmulators";
+import { Client } from "../apiv2";
+const fixturesDir = path.resolve(__dirname, "./fixtures");
+
+const ZIP_FIXTURES_PATH = path.join(fixturesDir, "zip-files");
+const ZIP_TEMPORARY_PATH = path.join(ZIP_FIXTURES_PATH, "temp");
+const ZIP_UNZIPPER_PATH = path.join(ZIP_FIXTURES_PATH, "node-unzipper-testData");
+
+describe("unzip", () => {
+  let uiZipPath: string;
+  let pubsubZipPath: string;
+  before(function (done) {
+    // eslint-disable-next-line @typescript-eslint/no-invalid-this
+    this.timeout(5000);
+    (async () => {
+      const [uiVersion, uiRemoteUrl] = [
+        DownloadDetails.ui.version,
+        DownloadDetails.ui.opts.remoteUrl,
+      ];
+      const [pubsubVersion, pubsubRemoteUrl] = [
+        DownloadDetails.pubsub.version,
+        DownloadDetails.pubsub.opts.remoteUrl,
+      ];
+
+      uiZipPath = path.join(ZIP_TEMPORARY_PATH, `ui-v${uiVersion}.zip`);
+      pubsubZipPath = path.join(ZIP_TEMPORARY_PATH, `pubsub-emulator-v${pubsubVersion}.zip`);
+
+      await fs.promises.mkdir(ZIP_TEMPORARY_PATH, { recursive: true });
+      if (!(await fs.promises.access(uiZipPath).catch(() => false))) {
+        await downloadFile(uiRemoteUrl, uiZipPath);
+      }
+
+      if (!(await fs.promises.access(pubsubZipPath).catch(() => false))) {
+        await downloadFile(pubsubRemoteUrl, pubsubZipPath);
+      }
+    })()
+      .then(done)
+      .catch(done);
+  });
+
+  after(async () => {
+    // await fs.promises.rmdir(ZIP_TEMPORARY_PATH, { recursive: true });
+  });
+
+  it("should unzip a ui emulator zip file", async function () {
+    // eslint-disable-next-line @typescript-eslint/no-invalid-this
+    this.timeout(2000);
+    await unzip(uiZipPath, path.join(ZIP_TEMPORARY_PATH, "ui"));
+
+    const files = await fs.promises.readdir(ZIP_TEMPORARY_PATH);
+    expect(files).to.include("ui");
+
+    const uiFiles = await fs.promises.readdir(path.join(ZIP_TEMPORARY_PATH, "ui"));
+    expect(uiFiles).to.include("client");
+    expect(uiFiles).to.include("server");
+
+    const serverFiles = await fs.promises.readdir(path.join(ZIP_TEMPORARY_PATH, "ui", "server"));
+    expect(serverFiles).to.include("server.js");
+  });
+
+  it("should unzip a pubsub emulator zip file", async function () {
+    // eslint-disable-next-line @typescript-eslint/no-invalid-this
```

Comment:
Can remove this linter violation by refactoring to:

it("should unzip...", async () => {}).timeout(2000);